src/main/java/org/folio/circulation/domain/Loan.java

```diff
@@ -255,6 +262,7 @@ public Loan overrideRenewal(DateTime dueDate,
 
   Loan checkIn(DateTime returnDate, UUID servicePointId) {
     changeAction("checkedin");
+    removeActionComment();
```

Comment:
Is the action comment showing upon checking in a feature or a bug? If a bug (as described by CIRC-251), is there a place where the comment was set that we can avoid setting on CheckIn, or only set at desired actions, rather than having an explicit "removeActionComment" call here?

Reply:
My understanding is that the `action comment` property is a comment for the last action performed (only used for overridden renewals, AFAIK).

I think the behaviour that CIRC-251 is trying to describe is that subsequent actions shouldn't have a repeated comment. It is a bug that they do.

That action history entries can have a comment is a feature.

Unfortunately, this is modelled as the removal of a comment, because of how `loan action history` was originally implemented, which has leaked into the loan-storage API and the domain of mod-circulation :-(

The loan action history, is actually made up of copies of the loan record upon each change (done by a trigger). 

This means that in order to get new values into it, we _have_ to introduce them into the loan record itself, e.g. `action` and `itemStatus` both of which are not real properties of a loan.

This is brittle, we will likely need to `remember` somehow that each action performed on a loan needs to clear the previous comment.

Does that make sense?

(cc: @mreno-EBSCO @skoczko @jeremythuff )

Reply:
@dmytrotur Does my reflection match your understanding?

Reply:
@marcjohnson-kint Yes, that is correct.

Reply:
Thanks for explaining, @marcjohnson-kint

Reply:
@mtraneis Does this explanation mean you are happy with this change?

Reply:
@marcjohnson-kint  yes, sorry if it wasn't clear :)